src/Compilers/Core/Portable/Operations/OperationExtensions.cs

```diff
@@ -10,6 +11,22 @@ namespace Microsoft.CodeAnalysis.Semantics
 {
     public static class OperationExtensions
     {
+        /// <summary>
+        /// This will check whether context around the operation has any error such as syntax or semantic error
+        /// </summary>
+        public static bool IsInvalid(this IOperation operation, Compilation compilation, CancellationToken cancellationToken = default(CancellationToken))
+        {
+            // once we made sure every operation has Syntax, we will remove this condition
+            if (operation.Syntax == null)
+            {
+                return true;
+            }
+
+            // if wrong compilation is given, GetSemanticModel will throw due to tree not belong to the given compilation.
+            var model = compilation.GetSemanticModel(operation.Syntax.SyntaxTree);
+            return model.GetDiagnostics(operation.Syntax.Span, cancellationToken).Any(d => d.Severity == DiagnosticSeverity.Error);
```

Comment:
This would mark operation being invalid if there's warnings that are treated as errors, is this intended? Id prefer to check for `Diagnostic.DefaultSeverity` instead.